src/components/project/project.service.ts

```diff
@@ -998,10 +998,40 @@ export class ProjectService {
       pendingBudget = budgets.items[0];
     }
 
+    const budgetPerms = await this.db
+      .query()
+      .optionalMatch([
+        node('user', 'User', { id: session.userId }),
+        relation('in', '', 'member'),
+        node('', 'SecurityGroup'),
+        relation('out', '', 'permission'),
+        node('canRead', 'Permission', {
+          property: 'budget',
+          read: true,
+        }),
+        relation('out', '', 'baseNode'),
+        node('project', 'Project', { id: project.id }),
+      ])
+      .optionalMatch([
+        node('user'),
+        relation('in', '', 'member'),
+        node('', 'SecurityGroup'),
+        relation('out', '', 'permission'),
+        node('canEdit', 'Permission', {
+          property: 'budget',
+          edit: true,
+        }),
+        relation('out', '', 'baseNode'),
+        node('project'),
+      ])
+      .raw('RETURN canRead.read as canRead, canEdit.edit as canEdit')
+      .first();
+
+    const budgetToReturn = current || pendingBudget;
     return {
-      value: current ? current : pendingBudget,
-      canEdit: true,
-      canRead: true,
+      value: budgetToReturn,
+      canRead: (budgetToReturn && budgetPerms?.canRead) || false,
+      canEdit: budgetPerms?.canEdit || false,
```

Comment:
Does canRead return true even when the list is empty? That to me is a sign that there's some difference between this query and the list query that needs to be rectified.

Reply:
yes i've been getting canRead true with an empty list.  I can try to investigate?  Not sure how/if they're related

Reply:
Just talked to @michaelmarshall, here's a summary:
The issue is as a Field Operations Director, when I crate a project, which creates a budget, the budget doesn't have the right permission path back to the creator so I don't have access to this budget.

This is the budget list query that's returning null (confirmed with Michael that it's the correct path, albeit too vague.  The perms Node should have canRead: true and property:budget):
`MATCH (User { id: '33UIVXUEzKD' })<-[:member]-(:SecurityGroup)-[:permission]->(perms:Permission)-[:baseNode]->(node:Budget)`

Need to investigate how the budget is created, but that's moved to java now